evento/src/views/EventPage/index.js

```diff
@@ -36,20 +54,46 @@ class EventPage extends Component {
 		});
 	}
 
+	onAttendButtonClick() {
+		const eventId = this.props.match.params.eventId;
+
+		fetch(`/events/${eventId}/attendees`, {
+			method: 'POST',
+			headers: { 'Authorization': Cookie.get('auth_token') }
+		})
+		.then(response => {
+			if (!response.ok) return Promise.reject();
+			// If successiful attending, re-fetch attendees
+			this.fetchAttendees();
+		})
+		.catch(() => this.setState({ errorMessage: "Something went wrong" }));
+	}
+
+	isUserAttending() {
+		let user = Cookie.get('user');
+		return JSON.stringify(this.state.attendees).includes(user);
```

Comment:
I think the following is clear and better than what you originally had. Everything makes sense in this code, where as in your code it is not at all clear that why JSON.stringify is required (or even that `user` is a JSON object).  

```
const userCookie = Cookie.get('user');
if(!userCookie) {
    return false;
}

const user = JSON.parse(userCookie);
return this.state.attendees.some(attendee => attendee.id == user.id);
```

Think about for an example a scenario where user has writes something to a text box, and you have to check whether it equals to some number in a variable. Your example is like the following:

```
string userInput = ...;
return this.Count.ToString() == userInput;
```

Sure, it works, but it is semantically incorrect. You are not comparing whether the userInput contains the number you are trying to match, but rather whether the it matches the string presentation of the number you are trying to match.

The following is just simply a more proper way to do it (although, it lacks some error handling):

```
string userInput = ...;
return this.Count == int.Parse(userInput);
```